dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java

```diff
@@ -815,4 +816,53 @@ public String getHTML(String uri, Host host, boolean liveMode, String contentId,
         } 
 
     }
+	
+    /**
+     * This returns the proper ihtml page based on id, state and language
+     * 
+     * @param id
+     * @param tryLang
+     * @param live
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+	@Override
+    public IHTMLPage findByIdLanguageFallback(final Identifier identifier, final long tryLang, final boolean live, final User user, final boolean respectFrontEndPermissions)
+            throws DotDataException, DotSecurityException {
+
+
+	    return findByIdLanguageFallback(identifier.getId(), tryLang, live, user, respectFrontEndPermissions);
+
+    }
+	
+	@Override
```

Comment:
shouldn't be annotated by CloseDbIfOpened

Reply:
I think so yes

Reply:
agreed